govuk-paas/paas-rds-broker
# Adding Read-Only user creation at binding.

What
-------
Our tenants asked for an ability to connect to the database instances that are managed by PaaS with a READ-ONLY user, to avoid changing the database by mistake, while troubleshooting for example.

We have changed the way we grant privileges to the created user during the binding of the database to the application. We are using custom parameter `read_only` (true|false) during `cf bind-service` command.

How to review
-------------------
- Code review
- Deploy CF to your DEV environment from latest master
- Make sure that internal apps with database are deployed and working - e.g. auditor, accounts, billing.
- Deploy CF to your DEV environment from the branch in (alphagov/paas-cf/pull/2417).
- Make sure pipeline is green
- bind to the existing db (auditor-db, accounts-db or billing-db) with another app (e.g.: static application will do).
  - deploy static-app
  - `cf bind-service static-app auditor-db -c '{"read_only": true}'`
- connect to the `auditor-db`  from bosh rds-broker instance with the read_only credentials (e.g.: you can get the creds via `cf env static-app`).
- connect to the `auditor-db` via conduit plugin - `cf conduit auditor-db -- psql`
  - create and populate a table in conduit session - check that you can access and read data from that table in the read-only session.
  -  create a new schema, table in that schema and populate the table in conduit session - check that you can access and read data from that schema/table in the read-only session.
  - create function in conduit session - check that you can user that function in the read-only session.

Who can review
---------------------
not @barsutka
